sirepo/package_data/static/js/raydata.js

```diff
@@ -616,6 +616,61 @@ SIREPO.app.directive('pngImage', function(plotting) {
     };
 });
 
+SIREPO.app.directive('columnPicker', function() {
+    return {
+        restrict: 'A',
+        scope: {
+            title: '@',
+            id: '@',
+            availableColumns: '=',
+            masterListColumns: '=',
+            columns: '=',
+            defaultColumns: '=',
+            setColumns: '=',
+            setAvailableColumns: '=',
+        },
+        template: `
+            <div class="modal fade" data-ng-attr-id="{{ id }}" tabindex="-1" role="dialog">
+                <div class="modal-dialog modal-lg">
+                    <div class="modal-content">
+                        <div class="modal-header bg-info">
+                            <button type="button" class="close" data-dismiss="modal"><span>&times;</span></button>
+                                <span class="lead modal-title text-info">{{ title }}</span>
+                            </div>
+                            <div class="modal-body">
+                                <div class="container-fluid">
+                                    <label for="scans-columns" style="margin-right: 10px;">Field:</label>
+                                        <select name="scans-columns" id="scans-columns" data-ng-model="selected" ng-change="selectColumn()">
+                                            <option ng-repeat="column in availableColumns">{{column}}</option>
+                                        </select>
+                                    </div>
+                                  </div>
+                                </div>
+                              </div>
+                            </div>
+        `,
+        controller: function($scope, appState, $interval, panelState) {
+            $scope.activeTab = 'basic';
+            $scope.selected = null;
+            $scope.setColumns();
+
+            $scope.selectColumn = function() {
+                if ($scope.selected !== null) {
+                    appState.models.metadataColumns.selected.push($scope.selected);
+                }
+                appState.saveChanges('metadataColumns');
+                $scope.setColumns();
```

Comment:
Pass in just one function that calls both of these.

Reply:
e64969d8a3a9303ebcf97566971db0411515341f